tests/models/test_gemini.py

```diff
@@ -981,6 +982,42 @@ async def test_image_url_input(allow_model_requests: None, gemini_api_key: str)
     assert result.output == snapshot("This is not a fruit; it's a pipe organ console.")
 
 
+@pytest.mark.vcr()
+async def test_video_as_binary_content_input(
+    allow_model_requests: None, gemini_api_key: str, video_content: BinaryContent
+) -> None:
+    m = GeminiModel('gemini-1.5-flash', provider=GoogleGLAProvider(api_key=gemini_api_key))
+    agent = Agent(m, system_prompt='You are a helpful chatbot.')
+
+    result = await agent.run(['Explain me this video', video_content])
+    assert (
+        result.output.strip()
+        == "That's a picture of a small, portable monitor attached to a camera, likely used for filming. The monitor displays a scene of a canyon or similar rocky landscape.  This suggests the camera is being used to film this landscape. The camera itself is mounted on a tripod, indicating a stable and likely professional setup.  The background is out of focus, but shows the same canyon as seen on the monitor. This makes it clear that the image shows the camera's viewfinder or recording output, rather than an unrelated display."
+    )
+
+
+@pytest.mark.vcr()
+async def test_video_url_input(allow_model_requests: None, gemini_api_key: str) -> None:
+    m = GeminiModel('gemini-1.5-flash', provider=GoogleGLAProvider(api_key=gemini_api_key))
+    agent = Agent(m, system_prompt='You are a helpful chatbot.')
+
+    video_url = VideoUrl(
+        url='https://data.grepit.app/data/offline/b9baaea5-79a2-4510-998b-d752e923af80/1903081239164404008/O4MkPE-UIe6oxq_f.mp4'
+    )
+
+    result = await agent.run(['Explain me this video', video_url])
+    assert result.output.strip() == (
```

Comment:
Can you please use the `snapshot` helper here and up above? It should pass the assertion just the same, but will be easier to update if we ever re-run the request (skipping VCR) and the output changes slightly.